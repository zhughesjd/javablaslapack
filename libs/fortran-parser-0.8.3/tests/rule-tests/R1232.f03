! Testing subroutine-stmt, R1232
11 SUBROUTINE foo
end subroutine
subroutine boo(a,b)
end subroutine
PURE subroutine coo()
end subroutine
subroutine doo() BIND(C)
end subroutine
end
