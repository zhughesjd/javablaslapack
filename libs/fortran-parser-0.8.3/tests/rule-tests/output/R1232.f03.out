R1232:subroutine-stmt__begin:
R1232:subroutine-stmt: label=[@2,33:34='11',<10>,2:0] keyword=[@4,36:45='SUBROUTINE',<155>,2:3] name=[@6,47:49='foo',<206>,2:14] eos=[@7,50:50='\n',<4>,2:17] hasPrefix=false hasDummyArgList=false hasBindingSpec=false hasArgSpecifier=false
R 204:specification-part: numUseStmts=0 numImportStmts=0 numDeclConstructs=0
R1234:end-subroutine-stmt: keyword1=[@8,51:53='end',<184>,3:0] keyword2=[@10,55:64='subroutine',<155>,3:4] name=null eos=[@11,65:65='\n',<4>,3:14]
R1232:subroutine-stmt__begin:
R1233:dummy-arg-list__begin:
R1233:dummy-arg: dummy=[@16,81:81='a',<206>,4:15]
R1233:dummy-arg: dummy=[@18,83:83='b',<206>,4:17]
R1233:dummy-arg-list: count=2
R1232:subroutine-stmt: keyword=[@12,66:75='subroutine',<155>,4:0] name=[@14,77:79='boo',<206>,4:11] eos=[@20,85:85='\n',<4>,4:19] hasPrefix=false hasDummyArgList=true hasBindingSpec=false hasArgSpecifier=true
R 204:specification-part: numUseStmts=0 numImportStmts=0 numDeclConstructs=0
R1234:end-subroutine-stmt: keyword1=[@21,86:88='end',<184>,5:0] keyword2=[@23,90:99='subroutine',<155>,5:4] name=null eos=[@24,100:100='\n',<4>,5:14]
R1232:subroutine-stmt__begin:
R1228:t-prefix-spec: spec=[@25,101:104='PURE',<143>,6:0]
R1227:t-prefix: specCount=1
R1232:subroutine-stmt: keyword=[@27,106:115='subroutine',<155>,6:5] name=[@29,117:119='coo',<206>,6:16] eos=[@32,122:122='\n',<4>,6:21] hasPrefix=true hasDummyArgList=false hasBindingSpec=false hasArgSpecifier=true
R 204:specification-part: numUseStmts=0 numImportStmts=0 numDeclConstructs=0
R1234:end-subroutine-stmt: keyword1=[@33,123:125='end',<184>,7:0] keyword2=[@35,127:136='subroutine',<155>,7:4] name=null eos=[@36,137:137='\n',<4>,7:14]
R1232:subroutine-stmt__begin:
R 509:language-binding-spec: keyword=[@43,155:158='BIND',<188>,8:17] language=[@45,160:160='C',<206>,8:22] hasName=false
R1232:subroutine-stmt: keyword=[@37,138:147='subroutine',<155>,8:0] name=[@39,149:151='doo',<206>,8:11] eos=[@47,162:162='\n',<4>,8:24] hasPrefix=false hasDummyArgList=false hasBindingSpec=true hasArgSpecifier=true
R 204:specification-part: numUseStmts=0 numImportStmts=0 numDeclConstructs=0
R1234:end-subroutine-stmt: keyword1=[@48,163:165='end',<184>,9:0] keyword2=[@50,167:176='subroutine',<155>,9:4] name=null eos=[@51,177:177='\n',<4>,9:14]
R1101:main-program__begin:
R 204:specification-part: numUseStmts=0 numImportStmts=0 numDeclConstructs=0
R1103:end-program-stmt: id=null
R1101:main-program: hasProgramStmt=false hasExecutionPart=false hasInternalSubprogramPart=false
